src/app/cli/src/cli_value_parser.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::str::FromStr;
+
+use internal_error::InternalError;
+use opendatafabric::{
+    DatasetName,
+    DatasetNamePattern,
+    DatasetRef,
+    DatasetRefAny,
+    DatasetRefPattern,
+    DatasetRefRemote,
+    Multihash,
+    RepoName,
+};
+use thiserror::Error;
+use url::Url;
+
+#[derive(Clone)]
+pub struct DatasetPatternValidationRes {
+    pub pattern: DatasetNamePattern,
+    pub is_dataset_ref: bool,
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[derive(Error, Debug)]
+pub enum ValidationDatasetRefError {
+    #[error(transparent)]
+    Failed(#[from] ValidationError),
+    #[error(transparent)]
+    Internal(
+        #[from]
+        #[backtrace]
+        InternalError,
+    ),
+}
+
+#[derive(Error, Debug)]
+#[error("{message}")]
+pub struct ValidationError {
+    pub message: String,
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+pub fn value_parse_dataset_ref_pattern_local(
+    s: &str,
+) -> Result<DatasetRefPattern, ValidationDatasetRefError> {
+    match DatasetRefPattern::from_str(s) {
+        Ok(drp) => Ok(drp),
+        Err(_) => Err(ValidationDatasetRefError::Failed(ValidationError {
+            message: "Local reference should be in form: `did:odf:...`, `my.dataset.id`, or a \
+                      wildcard pattern `my.dataset.%`"
+                .to_string(),
+        })),
+    }
+}
+
+pub fn value_parse_dataset_name(s: &str) -> Result<DatasetName, String> {
```

Comment:
Since these validators are only useful in CLI - we should use `pub(crate)` visibility.

`pub` should only be used for things we want to export from the crate.

If your `DatasetPatternValidationRes` type above was `pub(crate)` - compiler would let you know that its not used anywhere, otherwise it has to assume that someone who links `kamu-cli` as a library may use this type.